sources/core/Stride.Core.Mathematics/MathUtil.Matrix.cs

```diff
@@ -0,0 +1,97 @@
+using System.Numerics;
+using System.Transactions;
+
+namespace Stride.Core.Mathematics;
+public static partial class MathUtil
+{
+
+    public static Matrix4x4 Orthonormalize(Matrix4x4 matrix)
+    {
+        var result = matrix;
+
+        var row1 = new System.Numerics.Vector4(matrix.M11, matrix.M12, matrix.M13, matrix.M14);
```

Comment:
It is a bit unfortunate that `Matrix4x4` can't be read as 4 vectors, but it should be and you can do it from `M11`, `M21`, etc. There are several ways of doing that, e.g.:
```cs
var rows = MemoryMarshal.Cast<Matrix4x4, Vector4>(MemoryMarshal.CreateSpan(ref m, 1));
var row4 = rows[3]; // get the bounds check out of the way asap
var row3 = rows[2];
...
```
Or, even faster:
```
var row1 = Unsafe.As<float, Vector4>(ref m.M11);
var row2 = Unsafe.As<float, Vector4>(ref m.M21);
var row3 = Unsafe.As<float, Vector4>(ref m.M31);
var row4 = Unsafe.As<float, Vector4>(ref m.M41);
```
Could also use Bitcast, but since it is by value and the sizes don't match, I'd have to try. And either way not entirely sure the above code will be optimal.

Below you should do the reverse, since certainly 16 float assignments is a bad idea that you'd want to prevent everywhere you see them.